pkg/controller/volume/attachdetach/cache/actual_state_of_world.go

```diff
@@ -487,6 +492,27 @@ func (asw *actualStateOfWorld) addVolumeToReportAsAttached(
 	}
 }
 
+// AddNodeFromAttachUpdates add new node to nodesToUpdateStatusFor
+func (asw *actualStateOfWorld) AddNodeFromAttachUpdates(nodeName types.NodeName) {
+	asw.Lock()
+	defer asw.Unlock()
+	nodeToUpdate, nodeToUpdateExists := asw.nodesToUpdateStatusFor[nodeName]
+	if nodeToUpdateExists {
+		// should not happen
+		errMsg := fmt.Sprintf("Failed to add new node %q to nodesToUpdateStatusFor, "+
```

Comment:
It might happen, actually. Suppose a node is terminated and recreated, controller will get a new node add event. But the controller cache might still has the old node information which has the same name with new node